source/common/router/upstream_request.cc

```diff
@@ -223,6 +224,20 @@ const Network::Connection& UpstreamRequest::connection() const {
   return *parent_.callbacks()->connection();
 }
 
+void UpstreamRequest::copyDynamicMetaDataFromDownstream(StreamInfo::StreamInfo& l4_stream_info) {
```

Comment:
I think it'd be OK to have a "convention" over "configuration", but the way it's implemented is too specific to L4->L4 metadata transfer, so that leaves no room to generalize it later. So I see a bunch of options:
* reserve name for specific transfer, be very precise that it's L4 to L4;
* add API to upstream clusters, similar to internal upstream;
* add API to listeners to somehow access upstream metadata from filters;

Thoughts from others?

Reply:
I prefer to the first option. Do you have any suggestion about the name ?

Reply:
Shouldn't this also be done for tcp_proxy and other network routing filters?

Reply:
I think we can focus on the basic case for this PR.